src/ert/dark_storage/endpoints/parameters.py

```diff
@@ -72,13 +88,21 @@ def get_parameter_std_dev(
     *, storage: Storage = DEFAULT_STORAGE, ensemble_id: UUID, key: str, z: int
 ) -> Response:
     key = unquote(key)
-    ensemble = storage.get_ensemble(ensemble_id)
     try:
+        ensemble = storage.get_ensemble(ensemble_id)
         da = ensemble.calculate_std_dev_for_parameter_group(key)
     except ValueError as e:
+        logger.error(e)
         raise HTTPException(status_code=404, detail="Data not found") from e
+    except KeyError as ke:
+        logger.error(ke)
+        raise HTTPException(status_code=404, detail="Ensemble not found") from ke
+    except Exception as ex:
+        logger.exception(ex)
+        raise HTTPException(status_code=500, detail="Internal server error") from ex
 
     if z >= int(da.shape[2]):
+        logger.error("invalid z index")
```

Comment:
This error message feels like it could be a bit more precise and understandable for users.
Could you try to find an alternative?

Reply:
Has this been followed up?